lib/rubocop/cop/rails/root_join_chain.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Use a single `#join` instead of chaining on `Rails.root` or `Rails.public_path`.
+      #
+      # @example
+      #   # bad
+      #   Rails.root.join('db').join('schema.rb')
+      #   Rails.root.join('db').join(migrate).join('migration.rb')
+      #   Rails.public_path.join('path').join('file.pdf')
+      #   Rails.public_path.join('path').join(to).join('file.pdf')
+      #
+      #   # good
+      #   Rails.root.join('db', 'schema.rb')
+      #   Rails.root.join('db', migrate, 'migration.rb')
+      #   Rails.public_path.join('path', 'file.pdf')
+      #   Rails.public_path.join('path', to, 'file.pdf')
+      #
+      class RootJoinChain < Base
+        extend AutoCorrector
+
+        MSG = 'Use `Rails.%<root>s.join(...)` instead of chaining `#join` calls.'
+
+        RAILS_ROOTS = %i[root public_path].to_set.freeze
```

Comment:
Cosmetic: Does this need to be separate?

Reply:
Not sure what you mean with "separate". Should we make it a cop option, in case there are (or will be) more roots?

Reply:
I mean that it's extracted to a constant, but only used once. What do you think about inlining it to the node pattern?
          (send
            (send
              $(send
                (const {nil? cbase} :Rails) ${:root :public_path}) :join $...) :join $...)

Reply:
Done.